node/src/services/blockchain/handler.rs

```diff
@@ -72,19 +222,344 @@ impl ActorEventLoop<BlockchainService> for BlockchainServiceEventLoop {
+
+        let (result, rx) = self
+            .rpc_handlers
+            .rpc_query(&format!(
+                r#"{{
+                    "jsonrpc": "2.0",
+                    "method": "author_submitAndWatchExtrinsic",
+                    "params": ["0x{}"],
+                    "id": {:?}
+                }}"#,
+                array_bytes::bytes2hex("", &extrinsic.encode()),
+                array_bytes::bytes2hex("", &id_hash.as_bytes())
+            ))
+            .await
+            .expect("Sending query failed even when it is correctly formatted as JSON-RPC; qed");
+
+        let json: serde_json::Value =
+            serde_json::from_str(&result).expect("the result can only be a JSONRPC string; qed");
+        let error = json
+            .as_object()
+            .expect("JSON result is always an object; qed")
+            .get("error");
+
+        if let Some(error) = error {
+            return Err(anyhow::anyhow!("Error in RPC call: {}", error.to_string()));
+        }
+
+        Ok(RpcExtrinsicOutput {
+            hash: id_hash,
+            result,
+            receiver: rx,
+        })
+    }
+
+    /// Construct an extrinsic that can be applied to the runtime.
+    pub fn construct_extrinsic(
+        &self,
+        client: Arc<ParachainClient>,
+        function: impl Into<storage_hub_runtime::RuntimeCall>,
+        nonce: u32,
+    ) -> UncheckedExtrinsic {
+        let function = function.into();
+        let current_block_hash = client.info().best_hash;
+        let current_block = client.info().best_number.saturated_into();
+        let genesis_block = client
+            .hash(0)
+            .expect("Failed to get genesis block hash, always present; qed")
+            .expect("Genesis block hash should never not be on-chain; qed");
+        let period = BlockHashCount::get()
+            .checked_next_power_of_two()
+            .map(|c| c / 2)
+            .unwrap_or(2) as u64;
+        // TODO: Consider tipping the transaction.
+        let tip = 0;
+        let extra: SignedExtra = (
+        frame_system::CheckNonZeroSender::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckSpecVersion::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckTxVersion::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckGenesis::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckEra::<storage_hub_runtime::Runtime>::from(generic::Era::mortal(
+            period,
+            current_block,
+        )),
+        frame_system::CheckNonce::<storage_hub_runtime::Runtime>::from(nonce),
+        frame_system::CheckWeight::<storage_hub_runtime::Runtime>::new(),
+        pallet_transaction_payment::ChargeTransactionPayment::<storage_hub_runtime::Runtime>::from(
+            tip,
+        ),
+        cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim::<
+            storage_hub_runtime::Runtime,
+        >::new(),
```

Comment:
Out of curiosity, how does it work if you submit an extrinsic without these `extra` things (don't know what to call them).